src/auto-evo/EvolutionaryTree.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using AutoEvo;
+using Godot;
+using Godot.Collections;
+
+public class EvolutionaryTree : Control
+{
+    private const float TIMELINE_HEIGHT = 50.0f;
+    private const float TIMELINE_LINE_THICKNESS = 2.0f;
+    private const float TIMELINE_LINE_Y = 5.0f;
+    private const float TIMELINE_MARK_SIZE = 4.0f;
+    private const float SPECIES_SEPARATION = 50.0f;
+    private const float GENERATION_SEPARATION = 100.0f;
+    private const float SPECIES_NAME_OFFSET = 10.0f;
+
+    private readonly List<EvolutionaryTreeNode> nodes = new();
+
+    private readonly System.Collections.Generic.Dictionary<uint, string> speciesNames = new();
+
+    private readonly System.Collections.Generic.Dictionary<uint, EvolutionaryTreeNode> latestNodes = new();
+
+    private readonly System.Collections.Generic.Dictionary<uint, (uint ParentSpeciesID, int SplitGeneration)>
+        speciesOrigin = new();
+
+    private readonly ButtonGroup nodesGroup = new();
+
+    private Font latoSmall = null!;
+
+    private PackedScene treeNodeScene = null!;
+
+    private Vector2 treeNodeSize;
+
+    private uint maxSpeciesId;
+
+    private int latestGeneration;
+
+    [Signal]
+    public delegate void SpeciesSelected(int generation, uint id);
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        treeNodeScene = GD.Load<PackedScene>("res://src/auto-evo/EvolutionaryTreeNode.tscn");
+
+        var tempNode = treeNodeScene.Instance<EvolutionaryTreeNode>();
+        treeNodeSize = tempNode.RectSize;
+        tempNode.QueueFree();
+
+        latoSmall = GD.Load<Font>("res://src/gui_common/fonts/Lato-Italic-Small.tres");
+    }
+
+    public override void _Draw()
+    {
+        base._Draw();
+
+        // Draw timeline
+        DrawLine(new Vector2(0, TIMELINE_LINE_Y), new Vector2(RectSize.x, TIMELINE_LINE_Y), Colors.DarkCyan,
+            TIMELINE_LINE_THICKNESS, true);
+
+        for (int i = 0; i <= latestGeneration; i++)
+        {
+            DrawLine(new Vector2(i * GENERATION_SEPARATION + treeNodeSize.x / 2, TIMELINE_LINE_Y),
+                new Vector2(i * GENERATION_SEPARATION + treeNodeSize.x / 2, TIMELINE_LINE_Y + TIMELINE_MARK_SIZE),
+                Colors.DarkCyan, TIMELINE_LINE_THICKNESS, true);
+
+            var localizedText = i + " " + TranslationServer.Translate("MEGA_YEARS");
```

Comment:
That value is double, this one is int...

Reply:
That format also applies to ints as well, right? Because for Finnish for example it inserts spaces every 3 zeros in the number, just converting an int to a string, not even using the user's locale here, is not good enough.

Reply:
The fix you did here seems to have gotten reverted...